src/topology/algebra/monoid.lean

```diff
@@ -274,6 +275,55 @@ continuous.comp (continuous_pow n) h
 
 end has_continuous_mul
 
+section embedding
+
+variable [monoid M]
+
+@[to_additive]
+lemma function.injective_mul [monoid β] {f : β →* M} (h : function.injective f) (a b : β) :
+  a * b = h.to_local_equiv.symm ((f a) * (f b)) :=
```

Comment:
This lemma name doesn't describe the conclusion very well, although I can't think of a better name right now.